### DISCUSSION TITLE
Provision for Batch Numbers

### DISCUSSION THREAD

**Discussion by pkawalya** [drkNsubuga/PharmaSpot#31] [Ideas]
**Since the system is dealing with medicine, tracking medicine batch .**

**Comment by drkNsubuga** [comment]
Thanks for the suggestion. This is a planned feature which will be added in the next release. If you have any suggestions of how this can be implemented, feel free to make a pull request.

**Comment by pkawalya** [comment]
1. Medicine Batch Model
Fields:
batch_number (String): Unique identifier for each medicine batch.
medicine_name (String): Name of the medicine.
expiry_date (Date): Date the batch expires.
quantity (Integer): Number of units in the batch.
manufacturing_date (Date): Date the batch was manufactured.
supplier (String): Supplier from whom the medicine was purchased.
storage_conditions (String): Recommended storage conditions (e.g., temperature, humidity).
status (String): Current status of the batch (e.g., "In Stock", "Expired", "Distributed").

Model Relationship:
If a Medicine model exists, create a foreign key relationship between MedicineBatch and Medicine.

Each batch will be associated with a specific medicine type.

2. Medicine Batch Tracking
Batch Assignment:
When a medicine is added or distributed, associate it with a specific batch number in the database.

Batch Status Updates:
Add functionality to update the batch status as follows:
"In Stock": Available in inventory.
"Expired": The batch has passed the expiry date.
"Distributed": The batch has been distributed or used.

3. Inventory Management
Track Quantity per Batch:
Each batch should track the current quantity in stock.
Expiry Date and Low Stock Alerts:
Set up notifications or alerts to notify when:
A batch is nearing its expiry date.
The stock of a specific batch is running low.

4. Reporting
Batch Reports:
Include a report feature that shows:
Medicine batches sorted by batch number, expiry date, and current stock levels.
Highlight expired or low stock batches.

5. User Interface
Search Functionality:
Add a UI component where users can:
Search for medicine batches by batch number, status, or expiry date.
